src/index.js

```diff
@@ -1,54 +1,59 @@
 // Publish task
 var path = require('path'),
-  util = require('gulp-util'),
-  gift = require('gift'),
-  q = require('q'),
-  distRepoDir, distRepo, rootRepo, deferred, options = {};
+  util   = require('gulp-util'),
+  gift   = require('gift'),
+  fs     = require('fs'),
+  q      = require('q'),
+  distRepoDir, rootRepo, deferred, options = {};
 
 function parseOptions(options_) {
   options.folder = options_.folder || 'dist';
   options.branch = options_.branch || 'gh-pages';
-  options.cname = options_.cname || 'CNAME';
+  options.cname  = options_.cname  || 'CNAME';
+  options.domain = options_.domain || null;
   options.commit = options_.commit || 'Build '+(new Date());
-  options.cwd = options_.cwd || process.cwd();
+  options.cwd    = options_.cwd    || process.cwd();
   return options;
 }
 
 function fetchFromRootRepo() {
-  rootRepo.remote_fetch('-f ' + options.folder + '/ master:' + options.branch, function(err) {
-    if(err) throw err;
+  rootRepo.remote_fetch('-f ' + options.folder + '/ master:' + options.branch, function (err) {
+    if (err) throw err;
 
     deferred.resolve();
   });
 }
 
 function buildBranch(options_) {
-  parseOptions(options_);
+  var options = parseOptions(options_);
```

Comment:
At the moment `options` is global to the script, and `parseOptions` just populates it. See the end of line 7. It's like that so it can be used in utility functions like `fetchFromRootRepo`.

I've no objections to changing the way this works — but it should probably be consistent throughout.